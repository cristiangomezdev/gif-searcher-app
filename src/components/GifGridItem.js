import React from "react";
import PropTypes from 'prop-types';
export const GifGridItem = ({id,title,url})=>{
    return (
        <div className="card animate__animated animate__fadeIn animate__delay-0s"><img src={url} alt={title} key={id} ></img><p>{title}</p></div>)
}

GifGridItem.propTypes={
    id:PropTypes.string.isRequired,
    title:PropTypes.string.isRequired,
    url:PropTypes.string.isRequired,
 }