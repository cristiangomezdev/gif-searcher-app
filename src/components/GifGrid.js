import React from "react";
import { GifGridItem } from "./GifGridItem";
//import { PropTypes } from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
export const GifGrid = ({category})=>{
    const {data: images,loading} = useFetchGifs(category);

    return (<React.Fragment>
        <h3 className="animate__animated animate__fadeIn animate__delay-3s">{category}</h3>
        {loading && <p className="animate__animated animate__flash">Loading</p> }
        { <div className="card-grid">
        
            {images.map(image => <GifGridItem key={image.id} id={image.id} title={image.title} url={image.url} />)}
        </div> }
        </React.Fragment>)
}
//GifGrid.propTypes = { category: PropTypes.string.isRequired}