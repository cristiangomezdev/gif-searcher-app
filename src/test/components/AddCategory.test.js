import React from 'react';
import {shallow} from 'enzyme';
import { AddCategory } from '../../components/AddCategory';

describe('pruebas addcategory', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories={setCategories} />);

    beforeEach(()=>{
        jest.clearAllMocks();
         wrapper = shallow(<AddCategory setCategories={setCategories} />);
    })
    test('debe de mostrarse correctamente', () => {
        
        expect (wrapper).toMatchSnapshot();
    });

    test('Debe de cambiar la caja de texto', () => {
        const input = wrapper.find("input");
        const value = "Hola mundo";
        input.simulate("change", { target: { value } });
        const inputAfter = wrapper.find("input");
        expect(inputAfter.prop("value")).toBe(value);

    });
    test('No debe de postear la informacion con submit', () => {
        wrapper.find('form').simulate('submit',{preventDefault(){}})

        expect (setCategories).not.toHaveBeenCalled()
    });
    test('debe de llamar el setcategories y limpiar la caja de texto', () => {
        const value = 'hola mundo';
        wrapper.find("input").simulate("change", { target: { value } });
        wrapper.find('form').simulate('submit',{preventDefault(){}})
        expect( wrapper.find("input").text()).toEqual('')

    });
});