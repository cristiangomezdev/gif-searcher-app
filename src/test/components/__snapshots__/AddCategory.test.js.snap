// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas addcategory debe de mostrarse correctamente 1`] = `
<div>
  <h1 />
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</div>
`;
